### DISCUSSION TITLE
How do I filter C-H activation reactions?

### DISCUSSION THREAD

**Discussion by ruslankotl** [rdkit/rdkit#4947] [Q&A] [answered]
I am trying to filter out C–H borylation reactions out of a dataset via mapped SMILES strings. Since RDKit does not support atom mapping, I attempted [using atom mapping label as isotopes](https://github.com/rdkit/rdkit/issues/2367), however it didn't seem to work for me:

1.  I still could not specify a SMARTS query to retrieve [cH:1] to [c:1]B transformation without capturing Miyaura borylation C:1–X to C:1–B as well.
2. I do not understand how this would help if the isotope labels in query and in the reaction of interest will not match

**Comment by dehaenw** [comment, marked as answer]
hi @ruslankotl, 
one quick hack to filter out miyaura borylation is adding a filter that only retains reactions where the halogen count on the reactant and product side are the same. 
this function can be used to count the halogens (in this case F, Cl, Br and I)
```python
gethalogencount = lambda mol : sum(1 for atom in mol.GetAtoms() if atom.GetAtomicNum() in [9,17,35,53])
print(gethalogencount(Chem.MolFromSmiles("c1c(Cl)cccc1CF")))
```
returning `2`
however this will not deal with halogen equivalents (eg triflates) that also undergo miyaura borylation.
Another approach for your problem could be to take the products, define a reaction that removes the boronic group (boronic acids, pinacol boronates etc) and replaces it with H, and then comparing if the reactant from the DB and the product from the deboronation are the same.
